services/121-service/src/utils/token/token-validation.service.ts

```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@nestjs/common';
+import { TokenSet } from 'openid-client';
+
+@Injectable()
+export class TokenValidationService {
+  public isTokenValid(
+    tokenSet: TokenSet,
+    expiresAtInSeconds = false, // If true, then expires_at is assumed in milliseconds, as it should be
+  ): tokenSet is TokenSet & Required<Pick<TokenSet, 'access_token'>> {
+    if (!tokenSet || !tokenSet.expires_at) {
+      return false;
+    }
+
+    // convert to milliseconds if needed
+    if (expiresAtInSeconds) {
+      tokenSet.expires_at *= 1000;
+    }
+
+    const timeLeftBeforeExpire = tokenSet.expires_at - Date.now();
```

Comment:
What is not clear from the name of this function is that it looks at expires_at to determine if the token is still valid (it could also look at other params of TokenSet to determine this) and it is also not clear from the name of this function it looks if the token is still valid for at least 5 minutes.

Ideally we should name the function so that it is clearer what it does. Or else, GEFN, put a comment above the function with a brief explainer. 

See e.g. comment above IntersolveVisaService.retrieveAndUpdateWallet as an example of how we do function comments. In this way VS Code also picks it up and shows it when hovering.

Reply:
comment added